File: Task3/go/Subtask3/Task3.go
package main

import (
    "fmt"
)

func main() {
    var n int
    fmt.Print("Enter a number: ")
    fmt.Scan(&n)
	for i := 1; i <= n; i++ {
        for j := 0; j < n-i; j++ {
            fmt.Print(" ")
        }
        for k := 0; k < 2*i-1; k++ {
            fmt.Print("*")
        }
        fmt.Println()
    }
	for i := n - 1; i >= 1; i-- {
        for j := 0; j < n-i; j++ {
            fmt.Print(" ")
        }
        for k := 0; k < 2*i-1; k++ {
            fmt.Print("*")
        }
        fmt.Println()
    }
}

File: Task 03/haskell/Subtask3/Task3.hs
import Control.Monad (forM_)

printDiamond :: Int -> IO ()
printDiamond n = do
    let linesList = [1..n] ++ [n-1, n-2..1]
    forM_ linesList $ \i -> do
        let spaces = replicate (n - i) ' '
        let stars = replicate (2 * i - 1) '*'
        putStrLn (spaces ++ stars)

main :: IO ()
main = do
    putStr "Enter a number: "
    input <- getLine
    let n = read input :: Int
    printDiamond n
